File: frontend/src/api/scheduleApi.js
import api from './axiosInstance';

export const scheduleApi = {
  getAll: () => api.get('/schedules'),
  getById: (id) => api.get(`/schedules/${id}`),
  create: (data) => api.post('/schedules', data),
  update: (id, data) => api.put(`/schedules/${id}`, data),
  delete: (id) => api.delete(`/schedules/${id}`),
  getToday: () => api.get('/schedules/today'),
};

File: mobile/lib/api/schedule_api.dart
import 'api_client.dart';

class ScheduleApi {
  static Future<Map<String, dynamic>> getAll() => ApiClient.get('/schedules');
  static Future<Map<String, dynamic>> getToday() => ApiClient.get('/schedules/today');
  static Future<Map<String, dynamic>> getById(int id) => ApiClient.get('/schedules/$id');
  static Future<Map<String, dynamic>> create(Map<String, dynamic> data) => ApiClient.post('/schedules', body: data);
  static Future<Map<String, dynamic>> update(int id, Map<String, dynamic> data) => ApiClient.put('/schedules/$id', body: data);
  static Future<Map<String, dynamic>> delete(int id) => ApiClient.delete('/schedules/$id');
}
